web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java

```diff
@@ -60,6 +62,18 @@ public final class CookieServerCsrfTokenRepository implements ServerCsrfTokenRep
 
 	private int cookieMaxAge = -1;
 
+	private Consumer<ResponseCookie.ResponseCookieBuilder> cookieCustomizer = (builder) -> {};
+
+	/**
+	 * Add a {@link Consumer} for a {@code ResponseCookieBuilder} that will be invoked
+	 * for each cookie being built, just before the call to {@code build()}.
+	 * @param cookieCustomizer consumer for a cookie builder
+	 * @since 6.1
+	 */
+	public void setCookieCustomizer(Consumer<ResponseCookie.ResponseCookieBuilder> cookieCustomizer) {
+		this.cookieCustomizer = cookieCustomizer;
```

Comment:
Please check for null. You can use `Assert.notNull(cookieCustomizer, "cookieCustomizer must not be null")`.